src/argtypes.d

```diff
@@ -137,8 +140,95 @@ extern (C++) TypeTuple toArgTypes(Type t)
                     t.next.accept(this);
                     return;
                 }
+                // Zero-length arrays are always passed on the stack
+                else if (sz == 0)
+                {
+                    goto Lmemory;
+                }
+
+                const arraySize = t.size(Loc());
+
+                if (t.next.ty == Tfloat80)
+                {
+                    goto Lmemory;
+                }
+
+                if (arraySize < 1 || arraySize > 16)
+                {
+                    goto Lmemory;
+                }
+
+                const isFloat = t.next.isfloating();
+
+                switch (arraySize) {
+                    case 1:
+                        assert(0);
+                    case 2:
+                    case 3: // XXX not for m32 ?
```

Comment:
I don't know about DMD, but LDC uses `toArgTypes()` for the System V ABI only (i.e., non-Windows x86_64).